release notes/v0.42.0.md

```diff
@@ -0,0 +1,152 @@
+k6 v0.42.0 is here and it brings a few exciting updates! :tada:
+
+## Breaking changes
+
+- [#2712](https://github.com/grafana/k6/pull/2712) `k6/ws` returns an HTTP response for failed connections instead of an undefined behavior. Thanks, @brietaylor.
+
+## New Features
+
+### Experimental Prometheus Remote Write Output [#2784](https://github.com/grafana/k6/pull/2784)
+
+This release brings a new builtin Output to any [Prometheus Remote Write](https://docs.google.com/document/d/1LPhVRSFkGNSuU1fBd81ulhsCPR4hkSZyyBj1SZ8fWOM/edit) implementations (e.g. [Prometheus](https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver), [Mimir](https://grafana.com/docs/mimir/latest/operators-guide/reference-http-api/#remote-write)). This is an experimental feature, and future releases could introduce breaking changes.
+
+The following example uses `k6 run` with the new output. It uses the defaults options, such as the Remote Write server URL (http://localhost:9090/api/v1/write):
+
+```sh
+k6 run -o experimental-prometheus-rw script.js
+```
+
+It supports the new and convenient experimental [Native Histogram](https://prometheus.io/docs/practices/histograms) feature, added in Prometheus `v2.40.0`. It's not enabled by default, but we expect to make it the default way to map k6 [Trend metrics](https://k6.io/docs/javascript-api/k6-metrics/trend) once the Prometheus' project signals that its mature enough and when more Remote Write implementations support it. For now, if you want to use it, you need to set `K6_PROMETHEUS_RW_TREND_AS_NATIVE_HISTOGRAM` to `true`.
+
+You can find [complete documentation](https://k6.io/docs/results-output/real-time/prometheus-rw) with more examples, use cases, and available configurations.
+
+### More features for the experimental websockets module [#2786](https://github.com/grafana/k6/pull/2786) 
+
+The `k6/experimental/websockets` module that we announced in [v0.40.0](https://github.com/grafana/k6/releases/tag/v0.40.0) release got an update that extends its functionality.
```

Comment:
Let's link to the xk6-websockets repo.

And we didn't do this in the v0.40.0 release notes, but we should explain _why_ we're introducing an experimental WS module that mostly mimics what `k6/ws` does.

We should explain this, mentioning what our goal for the experimental module is (will it eventually replace `k6/ws`?), what the differences are, and the current state of the experimental module. Users will probably be wondering when/if they should start using it over `k6/ws`.

Maybe some of this is a better fit for the documentation itself (I've yet to review https://github.com/grafana/k6-docs/pull/933/), but we should at least mention here what the current status of the project is after these updates.

Reply:
>We should explain this, mentioning what our goal for the experimental module is (will it eventually replace k6/ws?), what the differences are, and the current state of the experimental module. Users will probably be wondering when/if they should start using it over k6/ws.

We should write this detailed info in the documentation then we could link it here. I think it is important enough to have it as permanent information and not in one specific release note. /cc @olegbespalov 

>Maybe some of this is a better fit for the documentation itself (I've yet to review https://github.com/grafana/k6-docs/pull/933), but we should at least mention here what the current status of the project is after these updates.

@olegbespalov can you help here?

Reply:
@olegbespalov I don't think we need to have a well-defined roadmap to make a short comment here about the project goals (replacing `k6/ws`?), and its current state (beyond just "experimental"; should people using `k6/ws` migrate to it, or people writing new WS scripts start using it instead of `k6/ws`?).

I think we missed the chance to mention the project goals when it was introduced in v0.40.0, so that can be done in the official documentation. But since we're announcing these updates here, we should at least expand on the status of it.

So maybe something like:

> This is still an experimental module, but with the recent changes we think it's usable for most users. So whether you're writing a new WebSocket test, or currently using the `k6/ws` module, we invite you to give it a try, and report any issues in the project's [issue tracker](https://github.com/grafana/xk6-websockets/issues).

Reply:
>  I don't think we need to have a well-defined roadmap to make a short comment here about the project goals (replacing k6/ws?)

Not sure that it's the right time to commit that it replace the `k6/ws`, but the suggested sentence looks great :heart: , so I'm up to add this to the current release notes. 

Do you want me to add this, or wanna do this by yourself?

Reply:
@olegbespalov I've added it. I think this now conflicts a bit with the warning in the documentation which it is very alarming. For example:

>use at your own risk
 
If possible, we should relax and make it more welcoming.

Reply:
> use at your own risk

It's a default warning that shows to all experimental extensions, I agree that it's a bit alarming, but probably it's better to tackle this separately, keeping in mind all the extensions.

Reply:
But I've tried to use @imiric 's words magic to make the landing page moe welcoming, for now, hope that works :crossed_fingers: